src/scib_metrics/_graph_connectivity.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+from scipy.sparse.csgraph import connected_components
+
+
+def graph_connectivity(X: csr_matrix, labels: np.ndarray) -> float:
+    """Quantify the connectivity of the subgraph per cell type label.
+
+    Parameters
+    ----------
+    X
+        Array of shape (n_cells, n_cells) with non-zero values
+        representing distances to exactly each cell's k nearest neighbors.
+    labels
+        Array of shape (n_cells,) representing label values
+        for each cell.
+    """
+    clust_res = []
+
+    for label in np.unique(labels):
+        mask = labels == label
+        graph_sub = X[mask]
+        graph_sub = graph_sub[:, mask]
+        _, comps = connected_components(graph_sub, connection="strong")
+        tab = pd.value_counts(comps)
+        clust_res.append(tab.max() / sum(tab))
+
+    return np.mean(clust_res)
```

Comment:
the cell types are weighted equally regardless of the proportions?

Reply:
yeah.... :(